app/gui2/src/components/ColorRing.vue

```diff
@@ -0,0 +1,187 @@
+<script setup lang="ts">
+import {
+  cssAngularColorStop,
+  gradientPoints,
+  rangesForInputs,
+} from '@/components/ColorRing/gradient'
+import { injectInteractionHandler } from '@/providers/interactionHandler'
+import { targetIsOutside } from '@/util/autoBlur'
+import { browserSupportsOklch, ensoColor, formatCssColor, parseCssColor } from '@/util/colors'
+import { Rect } from '@/util/data/rect'
+import { Vec2 } from '@/util/data/vec2'
+import { computed, onMounted, ref } from 'vue'
+
+/**
+ *  Hue picker
+ *
+ *  # Angles
+ *
+ *  All angles are measured in turns, starting from the 12-o'clock position, normalized to the range 0-1, unless
+ *  otherwise specified.
+ *  - This is the axis used by CSS gradients (adjustment is necessary when working with trigonometric functions, which
+ *    start from the positive x-axis).
+ *  - Turns allow constants to be expressed as simple numbers, and can be easily converted to the units used by external
+ *    APIs (radians for math, degrees for culori).
+ */
+
+// If the browser doesn't support OKLCH, the gradient will be specified by computing the number of points specified here
+// in OKLCH, converting to sRGB, and interpolating in HSL. This number has been found to be enough to look close to the
+// intended colors, without excessive gradient complexity (which may affect performance).
+const NONNATIVE_OKLCH_INTERPOLATION_STEPS = 300
```

Comment:
Is 300 really necessary? That sounds like a lot. Given that the fallback is still a gradient, I'd imagine just a few steps of rgb gradient should be really enough to decently approximate the gradient in proper color space.

Reply:
Much fewer steps are needed in sRGB.